lib/index.js

```diff
@@ -1,33 +1,48 @@
 var Webcast = require('webcast-osx-audio');
 var Client = require('castv2-client').Client;
 var DefaultMediaReceiver = require('castv2-client').DefaultMediaReceiver;
-var mdns = require('mdns');
+var chromecast = require('./chromecast');
 
 var Chromecast = function(options) {
   var deviceFound = false;
-  var browser = mdns.createBrowser(mdns.tcp('googlecast'));
+  
   if (options['l']) {
     console.log("Devices available for casting:");
   }
-  browser.on('serviceUp', function(service) {
 
-    if (typeof options['d'] !== 'undefined') {
-      if (options['d'] == service.name) {
-        // only start the service for the specified device.
-        ondeviceup(service.name, service.addresses[0]);
-      }
+  chromecast.search(function (err, devices) {
+    if (err) {
+
+      return;
     }
-    else {
-      if (!options['l']) {
-        console.log('found device "%s" at %s:%d', service.name, service.addresses[0], service.port);
+
+    if (devices) {
+      if (options['j']){
+        console.log(JSON.stringify(devices));
+        return;
+      }
+
+
+      for (var name in devices) {
```

Comment:
could this be updated either to a `devices.forEach(function(device) ...` or a `for (var i = 0 ...` loop. generally `for...in` should be avoided.